File: webui/src/test/java/com/taskadapter/config/ConfigStorageTest.java
package com.taskadapter.config;

import com.taskadapter.web.uiapi.ConfigId;
import com.taskadapter.web.uiapi.SetupId;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.TemporaryFolder;

import static org.assertj.core.api.Assertions.assertThat;

public class ConfigStorageTest {
    private final static String login = "autotest";

    @Rule
    public TemporaryFolder folder = new TemporaryFolder();
    private ConfigStorage storage;

    @Before
    public void beforeEachTest() {
        storage = new ConfigStorage(folder.getRoot());
    }

    @Test
    public void configCanBeCreatedInStorage() throws StorageException {
        var configId = createConfig(login);
        var config = storage.getConfig(configId);
        assertThat(config).isPresent();
        assertThat(config.get().getMappingsString()).isEqualTo("mappings");
    }

    @Test
    public void configIsDeleted() throws StorageException {
        var configId = createConfig(login);
        storage.deleteConfig(configId);
        assertThat(storage.getConfig(configId)).isNotPresent();
    }

    /**
     * regression test - a bug reported by user: NPE in getUserConfigs() where
     */
    @Test
    public void emptyStorageReturnsEmptyCollectionForGetUserConfigs() throws StorageException {
        createConfig(login);
        assertThat(storage.getUserConfigs(login)).isNotEmpty();
    }

    @Test
    public void configsAreLoadedByGetUserConfigs() throws StorageException {
        assertThat(storage.getUserConfigs(login)).isEmpty();
    }

    @Test
    public void getConfigsInFolderReturnsEmptyListForNullInput() {
        assertThat(storage.getConfigsInFolder(null)).isEmpty();
    }

    @Test
    public void getLegacyConfigsReturnsEmptyListForNullInput() {
        assertThat(storage.getLegacyConfigsInFolder("name", null, 1))
                .isEmpty();
    }

    private ConfigId createConfig(String login) throws StorageException {
        var connector1setupId = "web1";
        var connector2setupId = "web2";
        var configName = "some_config_name";
        return storage.createNewConfig(login, configName, "jira", new SetupId(connector1setupId), "value1",
                "jira", new SetupId(connector2setupId), "value2", "mappings");
    }
}

File: webui/src/test/scala/com/taskadapter/webui/config/ConfigStorageTest.scala
package com.taskadapter.webui.config

import com.taskadapter.config.ConfigStorage
import com.taskadapter.web.uiapi.SetupId
import com.taskadapter.webui.uiapi.ConfigsTempFolder
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}

@RunWith(classOf[JUnitRunner])
class ConfigStorageTest extends FunSpec with Matchers with ConfigsTempFolder {
  private val configName = "some_config_name"
  private val login = "autotest"

  it("config can be created in storage") {
    withTempFolder { folder =>
      val storage = new ConfigStorage(folder)
      val configId = storage.createNewConfig(login, configName, "jira", SetupId("web1"), "value1",
        "jira", SetupId("web2"), "value2", "mappings")
      val config = storage.getConfig(configId)
      config.isDefined shouldBe true
      config.get.getMappingsString shouldBe "mappings"
    }
  }

  it("config is deleted") {
    withTempFolder { folder =>
      val storage = new ConfigStorage(folder)
      val configId = storage.createNewConfig(login, configName, "jira", SetupId("web"), "value1",
        "jira", SetupId("web"), "value2", "mappings")
      storage.delete(configId)
      storage.getConfig(configId) shouldBe None
    }
  }
}
